cmd/status-cli/main.go

```diff
@@ -0,0 +1,347 @@
+						return err
+					}
+					accMsg := protocol.FindFirstByContentType(accRespAlice.Messages(), protobuf.ChatMessage_SYSTEM_MESSAGE_MUTUAL_EVENT_ACCEPTED)
+					fmt.Println("[Alice] got message:", accMsg.Text)
+
+					aliceContacts := aliceMessenger.MutualContacts()
+					fmt.Println("[Alice] contacts number:", len(aliceContacts))
+
+					// Send DM from alice to bob
+
+					aliceChat := aliceMessenger.Chat(aliceContacts[0].ID)
+					fmt.Println("[Alice] chat with contact id:", aliceChat.ID)
+
+					clock, timestamp := aliceChat.NextClockAndTimestamp(aliceMessenger.GetTransport())
+					inputMessage := common.NewMessage()
+					inputMessage.ChatId = aliceChat.ID
+					inputMessage.LocalChatID = aliceChat.ID
+					inputMessage.Clock = clock
+					inputMessage.Timestamp = timestamp
+					inputMessage.MessageType = protobuf.MessageType_ONE_TO_ONE
+					inputMessage.ContentType = protobuf.ChatMessage_TEXT_PLAIN
+					inputMessage.Text = "Hello Bob!"
+
+					chatResp, err := aliceMessenger.SendChatMessage(context.Background(), inputMessage)
+					fmt.Println("[Alice] function SendChatMessage response.messages:", chatResp.Messages())
+					if err != nil {
+						fmt.Println(err)
+						return err
+					}
+
+					chatRespBob, err := protocol.WaitOnMessengerResponse(
+						bobMessenger,
+						func(r *protocol.MessengerResponse) bool { return len(r.Messages()) > 0 },
+						"[Bob] not receive message from alice",
+					)
+					if err != nil {
+						fmt.Println(err)
+						return err
+					}
+					fmt.Println("[Bob] receive message from alice:", chatRespBob.Chats()[0].LastMessage.Text)
+
+					// Send DM from bob to alice
+
+					bobChat := bobMessenger.Chat(bobContacts[0].ID)
+					fmt.Println("[Bob] chat with contact id:", bobChat.ID)
+
+					bobClock, bobTimestamp := bobChat.NextClockAndTimestamp(bobMessenger.GetTransport())
+					bobInputMessage := common.NewMessage()
+					bobInputMessage.ChatId = bobChat.ID
+					bobInputMessage.LocalChatID = bobChat.ID
+					bobInputMessage.Clock = bobClock
+					bobInputMessage.Timestamp = bobTimestamp
+					bobInputMessage.MessageType = protobuf.MessageType_ONE_TO_ONE
+					bobInputMessage.ContentType = protobuf.ChatMessage_TEXT_PLAIN
+					bobInputMessage.Text = "Hello Alice!"
+
+					bobSendChatResp, err := bobMessenger.SendChatMessage(context.Background(), bobInputMessage)
+					fmt.Println("[Bob] function SendChatMessage response.messages:", bobSendChatResp.Messages())
+					if err != nil {
+						fmt.Println(err)
+						return err
+					}
+
+					aliceReceiveChatResp, err := protocol.WaitOnMessengerResponse(
+						aliceMessenger,
+						func(r *protocol.MessengerResponse) bool { return len(r.Messages()) > 0 },
+						"[Alice] not receive message from bob",
+					)
+					if err != nil {
+						fmt.Println(err)
+						return err
+					}
+					fmt.Println("[Alice] receive message from bob:", aliceReceiveChatResp.Chats()[0].LastMessage.Text)
+
+					return nil
+				},
+			},
+		},
+	}
```

Comment:
Can you please split this 5-levels code spaghetti into separate functions and variables? It's almost impossible to read.